### PR TITLE
Upgrade to v1.27.2

### PR DESCRIPTION
**Pull request by george-angel**

### CODE REVIEW

**Review by asiyani — APPROVED** [review]

File: resources/master-kubelet-conf.yaml [hunk 1]
```diff
@@ -15,6 +15,12 @@ cgroupDriver: systemd
 clusterDNS:${cluster_dns}
 clusterDomain: "cluster.local"
 ${feature_gates == "" ? "" : "featureGates:\n  ${feature_gates}"}
+containerRuntimeEndpoint: "unix:///run/containerd/containerd.sock"
+registerWithTaints:
+  - key: "node-role.kubernetes.io/master"
+    effect: "NoSchedule"
+  - key: "node-role.kubernetes.io/control-plane"
```
Comment (ffilippopoulos) [inline resources/master-kubelet-conf.yaml:22]: Everything scheduled on master nodes will need to be patched in advance to comply with that, like: https://github.com/utilitywarehouse/system-manifests/pull/188

File: variables.tf [hunk 2]
```diff
@@ -300,6 +300,8 @@ variable "containerd_no_shim" {
 }
 
 locals {
+  component_cloud_provider = var.cloud_provider == "aws" ? "external" : var.cloud_provider
```
[the remark this answers is not in the document]
Comment (ffilippopoulos) [reply variables.tf:303]: ahh yes, sorry for the spam. For some reason I thought I was looking at an aws only module here :facepalm:

**Review by ffilippopoulos — APPROVED** [review]

File: resources/kube-controller-manager.yaml [hunk 3]
```diff
@@ -12,7 +12,6 @@ spec:
       command:
         - kube-controller-manager
         ${cloud_config == "" ? "" : "- --cloud-config=/etc/kubernetes/config/cloud_provider/cloud.conf"}
-        ${cloud_provider == "" ? "" : "- --cloud-provider=${cloud_provider}"}
```
Comment (ffilippopoulos) [inline resources/kube-controller-manager.yaml:15]: I do not think we are doing this correctly. Reading through this: https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/ I think that there are 2 steps for kube-controller-manager to migrate to external cloud providers.
- First, you just roll the flag `--enable-leader-migration` to kube-controller-manager pods of kube version 1.26.
- Then, you change `--cloud-provider` to `external` as you roll to the latest kube and have already deployed the external cloud-controller.

Comment (george-angel) [reply resources/kube-controller-manager.yaml:15]: That does sound much better - rollout the external providers on 1.26 and then roll cleanly onto 1.27

Comment (ffilippopoulos) [reply resources/kube-controller-manager.yaml:15]: not quite. Reading again and again:

> The control plane now contains nodes of both version N and N + 1. The nodes of version N run kube-controller-manager only, and these of version N + 1 run both kube-controller-manager and cloud-controller-manager. The migrated controllers, as specified in the configuration, are running under either kube-controller-manager of version N or cloud-controller-manager of version N + 1 depending on which controller manager holds the migration lease. No controller will ever be running under both controller managers at any time.

I think that the cloud-controller has to come with the update and kube-controller-manager will know to not compete for the lease during the migration.

**Pull request merged**